google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -15,161 +15,382 @@
 # limitations under the License.
 #
 
-from typing import Dict, Union, Optional
 
+from typing import Dict, Union, Optional, Any
+
+from google.api_core import exceptions
+from google.auth import credentials as auth_credentials
+from google.protobuf import timestamp_pb2
+
+from google.cloud.aiplatform import base
+from google.cloud.aiplatform import gapic
+from google.cloud.aiplatform import pipeline_jobs
+from google.cloud.aiplatform.compat.types import execution as gca_execution
 from google.cloud.aiplatform.metadata import constants
-from google.cloud.aiplatform.metadata.artifact import _Artifact
-from google.cloud.aiplatform.metadata.context import _Context
-from google.cloud.aiplatform.metadata.execution import _Execution
-from google.cloud.aiplatform.metadata.metadata_store import _MetadataStore
+from google.cloud.aiplatform.metadata import context
+from google.cloud.aiplatform.metadata import execution
+from google.cloud.aiplatform.metadata import experiment_resources
+from google.cloud.aiplatform.metadata import experiment_run_resource
+from google.cloud.aiplatform.tensorboard import tensorboard_resource
+
+_LOGGER = base.Logger(__name__)
+
+
+def _get_experiment_schema_version() -> str:
+    """Helper method to get experiment schema version
 
+    Returns:
+        str: schema version of the currently set experiment tracking version
+    """
+    return constants.SCHEMA_VERSIONS[constants.SYSTEM_EXPERIMENT]
 
+
+# Legacy Experiment tracking
+# Maintaining creation APIs for backwards compatibility testing
 class _MetadataService:
     """Contains the exposed APIs to interact with the Managed Metadata Service."""
 
+    @staticmethod
```

Comment:
Same comment as with metadata_store.py:

It's not directly in this PR, but what's the rationale for making this class non-public?

There's a public metadata client for it. Are there no scenarios where a user might want to manage metadata via the SDK?

Reply:
This interface wasn't proposed during design and needs additional considerations and review. The Metadata team is working in parallel to enable additional Metadata CUJs in the SDK.